decompiler/frontend/binaryninja/tagging.py

```diff
@@ -0,0 +1,32 @@
+import binaryninja
+from binaryninja import TagType
+from compiler_idioms.matcher import Matcher
+
+
+class CompilerIdiomsTagging:
+    TAG_SYMBOL = "⚙"
+
+    def __init__(self, binary_view: binaryninja.BinaryView, start: int):
+        self._bv = binary_view
+        self._path = self._bv.file.filename
+        self._function_start = start
+
+    def run(self):
+        matches = Matcher().find_idioms_in_function(self._path, self._function_start)
+        for match in matches:
+            for address in match.addresses:
+                self._set_tag(self._bv, tag_name=f"compiler_idiom: {match.operation}", address=address,
+                              text=f"{match.operand},{match.constant}")
+
+    @staticmethod
+    def _set_tag(binary_view: binaryninja.BinaryView, tag_name: str, address: int, text: str):
+        if binary_view.get_user_data_tags_at(address):
```

Comment:
I would prefer to check if there is a tag of the specified symbol rather than checking for ANY tag. Otherwise, you risk not tagging an idiom if there already is a different tag, right?

Reply:
To be honest, the better solution would probably be to not use Binary Ninjas Tagging system anymore. However, I understand that this would require far more adaptions for the tagging as well as the processing in the corresponding later stage.